src/Google/GlobalSiteTag.php

```diff
@@ -172,4 +418,21 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
 	public static function is_needed(): bool {
 		return true;
 	}
+
+	/**
+	 * Check if it is the new customer order.
+	 *
+	 * @param string $customer_email Customer email address.
+	 * @return bool True if this is new customer order.
+	 */
+	private static function is_first_time_customer( $customer_email ): bool {
+		$query = new \WC_Order_Query(
+			[
+				'return' => 'ids',
+			]
+		);
+		$query->set( 'customer', $customer_email );
+		$orders = $query->get_orders();
+		return count( $orders ) === 1 ? 'true' : 'false';
```

Comment:
Since we are only interested in knowing if there is at least 1 order with the matching email. We can limit the query to only one result, so it can stop scanning the DB table after finding the first match. For large sites searching customer orders can potentially be slow.

Reply:
Great suggestion! I updated it to 2 here.